Samples/custom-transform-adapter/server.go

```diff
@@ -0,0 +1,284 @@
+}
+
+func (client *BridgeClientAutorest) SendMessage(ctx context.Context, deviceID string, body *bridge.MessageBody) (autorest.Response, error) {
+	return client.BaseClient.SendMessage(ctx, deviceID, body)
+}
+
+func (client *BridgeClientAutorest) GetBaseURI() string {
+	return client.BaseURI
+}
+
+type Adapter struct {
+	GetBridgeClient func() BridgeClient
+	Router          *mux.Router
+	Engine          *TransformEngine
+}
+
+// D2C message route definition augmented to include the Id of the cached transform queries.
+type AugmentedD2CMessage struct {
+	D2CMessage
+	TransformId         string
+	DeviceIdBodyQueryId string
+}
+
+// Builds a transform adapter for a given configuration.
+func NewAdapterFromConfig(config *Config, bridgeEndpoint string) *Adapter {
+	log.Info(fmt.Sprintf("Initializing adapter for Bridge %s", bridgeEndpoint))
+
+	adapter := Adapter{
+		Engine: NewTransformEngine(),
+		Router: mux.NewRouter(),
+		GetBridgeClient: func() BridgeClient {
+			return &BridgeClientAutorest{bridge.NewWithBaseURI(bridgeEndpoint)}
+		},
+	}
+
+	for _, message := range config.D2CMessages {
+		log.Info(fmt.Sprintf("Initializing route %s", message.Path))
+		augmentedMessage := AugmentedD2CMessage{D2CMessage: message}
+
+		// Initialize cache for request body transform.
+		if message.Transform != "" {
+			augmentedMessage.TransformId = uuid.New().String()
+			if err := adapter.Engine.AddTransform(augmentedMessage.TransformId, message.Transform); err != nil {
+				log.WithField("error", err).Panic(fmt.Sprintf("Failed to add request body transform for route %s: %s", message.Path, err))
+			}
+		} else {
+			log.Warn(fmt.Sprintf("Empty transform. Route %s will be set as pass-through", message.Path))
+		}
+
+		// Initialize cache for device Id transform.
+		if message.DeviceIdBodyQuery != "" {
+			augmentedMessage.DeviceIdBodyQueryId = uuid.New().String()
+			if err := adapter.Engine.AddTransform(augmentedMessage.DeviceIdBodyQueryId, message.DeviceIdBodyQuery); err != nil {
+				log.WithField("error", err).Panic(fmt.Sprintf("Failed to add device Id query transform for route %s: %s", message.Path, err))
+			}
+		}
+
+		handler := adapter.buildD2CMessageHandler(augmentedMessage)
+		adapter.Router.HandleFunc(message.Path, withLogging(handler)).Methods("POST")
+	}
+
+	return &adapter
+}
+
+func (adapter *Adapter) ListenAndServe(port int) error {
+	log.Info(fmt.Sprintf("Server listening on port %d", port))
+	return http.ListenAndServe(fmt.Sprintf(":%d", port), adapter.Router)
+}
+
+// Builds the HTTP handler for a given C2D route definition.
+func (adapter *Adapter) buildD2CMessageHandler(message AugmentedD2CMessage) func(*log.Entry, http.ResponseWriter, *http.Request) {
+	return func(logger *log.Entry, w http.ResponseWriter, r *http.Request) {
+		var jsonBody map[string]interface{}
+		if err := decodeJsonBody(w, r, &jsonBody); err != nil {
+			respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Failed to decode JSON body: %s", err.Error()))
```

Comment:
nit: i'd probably make this an actual `error`. you can do this:
```suggestion
			respondError(logger, w, http.StatusBadRequest, fmt.Errorf("Failed to decode JSON body: %w", err))
```